content/en/docs/components/pipelines/compile-a-pipeline.md

```diff
@@ -0,0 +1,123 @@
++++
+title = "Compile a Pipeline"
+description = "Compile a pipeline definition to YAML"
+weight = 5
++++
+
+While pipelines and components are authored in Python, they can be compiled to intermediate representation (IR) YAML ([example][compiled-output-example]).
+
+A YAML pipeline/component definition preserves a static representation of your pipeline/component. This YAML can be submitted to the KFP backend for execution or deserialized by the KFP SDK for integration into another pipeline.
+
+## Compiling
+First let's define a very simple pipeline:
+
+```python
+from kfp import compiler
+from kfp import dsl
+
+@dsl.component
+def addition_component(num1: int, num2: int) -> int:
+    return num1 + num2
+
+@dsl.pipeline(name='addition-pipeline')
+def my_pipeline(a: int, b: int, c: int = 10):
+    add_task_1 = addition_component(num1=a, num2=b)
+    add_task_2 = addition_component(num1=add_task_1.output, num2=c)
+```
+
+Now we can compile the pipeline to the file `my_pipeline.yaml`:
+
+```python
+cmplr = compiler.Compiler()
+cmplr.compile(my_pipeline, package_path='my_pipeline.yaml')
+```
+
+Just as a pipeline is a template for a multi-step workflow, a component is a template for a single-step workflow. We can also compile the component `addition_component` directly:
+
+```python
+cmplr.compile(addition_component, package_path='addition_component.yaml')
+```
+
+The `Compiler.compile` method accepts a few optional additional parameters:
+
+**`pipeline_name`** (`string`)
+
+Sets the name of the pipeline (or component). This is written to IR as the  `pipelineInfo.name` field. Will override the `name` passed to the `@dsl.pipeline` decorator.
```

Comment:
nit: mention this name is used a pipeline context and explains a bit what is pipeline context?

Reply:
By pipeline context, are you referring to the pipeline context manager we use to create pipeline spec? If so, isn't that largely an implementation detail and not something we need to explain to authors? If not, can let me know what you are referring to?

Reply:
No, pipeline context is maybe a MLMD concept. It identifies a number of runs originated from the same pipeline definition. I think it provides a query scope for some features like caching and importer. Desmond and/or Jialu may provide better explanation on this.